File: Restful API/Retrofit/app/src/main/java/com/okidoki/android/retrofit/dog_app/java/activity/DetailDogActivity.java
package com.okidoki.android.retrofit.dog_app.java.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import com.okidoki.android.retrofit.R;
import com.okidoki.android.retrofit.dog_app.java.adapter.DogAdapter;
import com.okidoki.android.retrofit.dog_app.java.data.DogImages;
import com.okidoki.android.retrofit.dog_app.java.service.RetrofitService;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class DetailDogActivity extends AppCompatActivity {

    private String breedName;
    private RecyclerView mRecyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail_dog);

        mRecyclerView = findViewById(R.id.recycler_view);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
    }

    @Override
    protected void onResume() {
        super.onResume();
        breedName = getIntent().getStringExtra(MainDogActivity.BREED_KEY);

        if (breedName == null || breedName.isEmpty() ) {
            Log.e("DetailActivity", "selectedBreed is null");
            return;
        }
        callRxRetrofit(breedName);
    }

    private void callRxRetrofit(String searchString){
        Observable<DogImages> result = RetrofitService.Factory.create(searchString);

        result
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Observer<DogImages>() {
                    @Override
                    public void onSubscribe(Disposable d) {
                    }

                    @Override
                    public void onNext(DogImages value) {
                        bindAdapterToView(value.message);
                    }

                    @Override
                    public void onError(Throwable e) {
                    }

                    @Override
                    public void onComplete() {
                    }
                });
    }

    private void bindAdapterToView(List<String> list){
        mRecyclerView.setAdapter(new DogAdapter(list, this));
    }
}

File: Restful API/Retrofit/app/src/main/java/com/okidoki/android/retrofit/dog_app/kotlin/activity/DetailDogActivity.kt
package com.okidoki.android.retrofit.dog_app.kotlin.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.widget.Toast
import com.okidoki.android.retrofit.R
import com.okidoki.android.retrofit.dog_app.java.activity.MainDogActivity.BREED_KEY
import com.okidoki.android.retrofit.dog_app.kotlin.adapter.DogAdapter
import com.okidoki.android.retrofit.dog_app.kotlin.service.DogService
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_detail_dog.*

class DetailDogActivity : AppCompatActivity() {
    private var disposable: Disposable? = null

    private val dogService by lazy {
        DogService.create()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail_dog)

        var breedName = intent.getStringExtra(BREED_KEY)
        recycler_view.setLayoutManager(LinearLayoutManager(this))

        startSearch(breedName);
    }

    override fun onPause() {
        super.onPause()
        disposable?.dispose()
    }

    private fun startSearch(searchString: String) {
        disposable = dogService.getDogImages(searchString)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                        { result -> if (result.message != null) bindAdapterToView(result.message!!) },
                        { error -> Toast.makeText(this, error.message, Toast.LENGTH_SHORT).show() }
                )
    }

    fun bindAdapterToView(list: List<String>){
        recycler_view.setAdapter(DogAdapter(list,this))
    }
}